beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -30,24 +32,38 @@ enum RangeBlockDataRequest<E: EthSpec> {
             DataColumnsByRangeRequestId,
             ByRangeRequest<DataColumnsByRangeRequestId, DataColumnSidecarList<E>>,
         >,
+        /// The column indices corresponding to the request
+        column_peers: HashMap<DataColumnsByRangeRequestId, Vec<u64>>,
         expected_custody_columns: Vec<ColumnIndex>,
     },
 }
 
+#[derive(Debug)]
+pub struct CouplingError {
+    pub(crate) msg: String,
+    pub(crate) column_and_peer: Option<(Vec<(u64, PeerId)>, PeerAction)>,
+}
+
 impl<E: EthSpec> RangeBlockComponentsRequest<E> {
+    #[allow(clippy::type_complexity)]
     pub fn new(
         blocks_req_id: BlocksByRangeRequestId,
         blobs_req_id: Option<BlobsByRangeRequestId>,
-        data_columns: Option<(Vec<DataColumnsByRangeRequestId>, Vec<ColumnIndex>)>,
+        data_columns: Option<(
+            Vec<(DataColumnsByRangeRequestId, Vec<u64>)>,
```

Comment:
nit - for readability:

```suggestion
            Vec<(DataColumnsByRangeRequestId, Vec<ColumnIndex>)>,
```